src/code.c

```diff
@@ -921,13 +921,34 @@ void CodeIfElse ( void )
 
 void CodeIfBeginBody ( void )
 {
+    // get and check the condition
+    Expr cond = PopExpr();
+
+    // if the condition is 'false', ignore the body
+    if (TNUM_EXPR(cond) == T_FALSE_EXPR) {
+        STATE(IntrIgnoring) = 1;
+    }
+    else {
+        // put the condition expression back on the stack
+        PushExpr(cond);
+    }
 }
 
-void CodeIfEndBody (
+Int CodeIfEndBody (
     UInt                nr )
 {
     /* collect the statements in a statement sequence if necessary         */
     PushStat( PopSeqStat( nr ) );
+
+    // get and check the condition
+    Expr cond = PopExpr();
+    PushExpr(cond);
+
+    // if the condition is 'true', ignore other branches of the if-statement
+    if (TNUM_EXPR(cond) == T_TRUE_EXPR) {
+        STATE(IntrIgnoring) = 1;
```

Comment:
One thing I don't like about this PR is that I have to access `IntrIgnoring` twice in `src/code.c`; previously, there was no other access to `IntrIgnoring`, `IntrReturning` or `IntrCoding` in `src/code.c`. And only a handful of access to any of them outside of `src/intrprt.c` (namely in `src/gap.c` and `src/read.c`).

But this is only a minor blemish, IMHO